astroquery/mast/core.py

```diff
@@ -683,8 +695,8 @@ def service_request_async(self, service, params, pagesize=None, page=None, **kwa
             mashup_request[prop] = value
 
         req_string = _prepare_service_request_string(mashup_request)
-        response = self._request("POST", self._MAST_REQUEST_URL, data=req_string, headers=headers,
-                                 retrieve_all=retrieve_all)
+        response = self._request_w_cache("POST", self._MAST_REQUEST_URL, data=req_string, headers=headers,
```

Comment:
no need for this, pass `cache` to `self._request`, it should know about it (coming from the BaseQuery class)